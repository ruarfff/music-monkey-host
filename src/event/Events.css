.eventCreateActionContainer {
  align-items: center;
  display: flex;
  flex-flow: row wrap;
  justify-content: space-evenly;
  padding-bottom: 1.5em;
}
.eventCreateButton {
  float: right;
  margin: 1em;
  text-align: center;
}

.eventCreateText {
  margin: 1em;
  text-align: center;
}

.events {
  align-items: center;
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
  margin-top: 1em;
  width: 100%;
}

.eventsList {
  display: flex;
  flex-flow: row wrap;
  height: 100%;
  justify-content: flex-start;
  margin-top: 2em;
  width: 100%;
}

.eventsListCaption {
  text-align: center;
  width: 100%;
}

.eventsUpcoming {
  border-right: 1px solid;
}
