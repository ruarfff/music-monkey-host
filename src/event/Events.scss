.eventCreateActionContainer {
  align-items: center;
  display: flex;
  flex-flow: row wrap;
  justify-content: space-evenly;
  padding-bottom: 1.5em;
}
.eventCreateButton {
  float: right;
  margin: 1em;
  text-align: center;
}

.eventCreateText {
  margin: 1em;
  text-align: center;
}

.events {
  align-items: center;
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
  margin-top: 1em;
  width: 100%;
  padding: 20px;
}

.slider-frame {
  overflow: visible!important;
  min-height: 200px;
}

.slider-list {
  max-height: 210px!important;
}

.slider-control-bottomcenter {
  display: none!important;
}

.slider {
  overflow: hidden;
  padding-left: 40px;
  padding-bottom: 2px;
}

.eventsList {
  display: flex;
  flex-flow: row wrap;
  height: 100%;
  justify-content: flex-start;
  width: 100%;
}

.eventListTitle {
  font-family: 'Roboto', 'sans-serif';
  font-size: 20px;
  line-height: 23px;
  color: black;
  font-weight: 500;
  margin-right: 10px;
}

.eventListShowAll {
  font-family: 'Roboto', sans-serif;
  font-size: 14px;
  line-height: 16px;
  color: #FFB000;
  font-weight: 500;
  text-transform: uppercase;
  text-decoration: #AF00FF;
  letter-spacing: 0.75px;
}

.eventsRow {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  height: 220px;
}

.eventCardIcon {
  margin-right: 14px;
}
