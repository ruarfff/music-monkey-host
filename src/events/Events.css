.Events {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  flex-wrap: wrap;
  margin-top: 1.5em;
  width: 100%;
}

.Events-no-events-message {
  text-align: center;
}

.Events-list-caption {
  text-align: center;
  width: 100%;
}

.Events-list {
  display: flex;
  flex-flow: row wrap;
  justify-content: space-evenly;
  margin-top: 4em;
}

.Events-upcoming {
  border-right: 1px solid;
}

.Event-create-action-container {
  display: flex;
  flex-flow: row wrap;
  justify-content: space-evenly;
  align-items: center;
  padding-bottom: 2em;
}

.Event-create-text {
  float: left;
  margin: 1em;
  text-align: center;
}
.Event-create-button {
  float: right;
  margin: 1em;
  text-align: center;
}