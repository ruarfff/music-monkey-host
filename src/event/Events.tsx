import Divider from '@material-ui/core/Divider/Divider'
import Grid from '@material-ui/core/Grid/Grid'
import Hidden from '@material-ui/core/Hidden/Hidden'
import IconButton from '@material-ui/core/IconButton'
import Typography from '@material-ui/core/Typography/Typography'
import _ from 'lodash'
import { map, sortBy } from 'lodash'
import moment from 'moment'
import Carousel from 'nuka-carousel'
import * as React from 'react'
import { Link } from 'react-router-dom'
import arrowLeft from '../assets/arrow-left.svg'
import arrowRight from '../assets/arrow-right.svg'
import IAction from '../IAction'
import LoadingSpinner from '../loading/LoadingSpinner'
import IUserState from '../user/IUserState'
import EventCard from './EventCard'
import './Events.scss'
import IEvent from './IEvent'
import IEventState from './IEventState'
import NoEvents from './NoEvents'
import PlaylistCard from './PlaylistCardSmall'

interface IEventsProps {
  events: IEventState
  user: IUserState
  getEvents(): IAction
}

class Events extends React.Component<IEventsProps> {
  public componentDidMount() {
    if (this.props.user.data) {
      this.props.getEvents()
    }
  }

  public renderCarousel = (
    events: IEvent[],
    playlist: boolean,
    message?: string
  ) => {
    const now = moment()

    let upcomingPlaylists: any[] = []

    if (!!events && playlist) {
      upcomingPlaylists = _.uniqBy(
        events
          .filter(event => event.startDateTime.isAfter(now))
          .map(event => event.playlist),
        'id'
      )
    }

    const leftControl = ({ previousSlide }: any) => (
      <IconButton onClick={previousSlide}>
        <img src={arrowLeft} alt="left" />
      </IconButton>
    )

    const rightControls = ({ nextSlide }: any) => (
      <IconButton onClick={nextSlide}>
        <img src={arrowRight} alt="right" />
      </IconButton>
    )

    return (
      <React.Fragment>
        {!playlist && events.length === 0 && <NoEvents />}
        {playlist &&
          events.length === 0 && (
            <Typography
              className="eventsListCaption"
              align="center"
              variant="body2"
              gutterBottom={true}
            >
              {message}
            </Typography>
          )}
        {events.length > 0 && (
          <Carousel
            slidesToShow={4}
            slidesToScroll={1}
            renderCenterLeftControls={leftControl}
            renderCenterRightControls={rightControls}
            withoutControls={
              (
                (!(events.length > 5) && !playlist) ||
                (!(upcomingPlaylists.length > 5) && playlist)
              ) && true
            }
          >
            {events.length > 0 &&
              !playlist &&
              map(
                sortBy(
                  events,
                  (event: IEvent) => event.startDateTime
                ).reverse(),
                (event: IEvent) => (
                  <EventCard key={event.eventId} event={event} />
                )
              )}
            {playlist &&
              upcomingPlaylists.map((upcomingPlaylist, index) => (
                <PlaylistCard key={index} playlist={upcomingPlaylist} />
              ))}
          </Carousel>
        )}
      </React.Fragment>
    )
  }

  public render() {
    const { events, eventsLoading } = this.props.events
    const now = moment()
    let upcomingEvents: IEvent[] = []
    if (!!events) {
      upcomingEvents = events.filter(event => event.startDateTime.isAfter(now))
    }

    return (
      <div className="events">
        {eventsLoading && <LoadingSpinner />}

        {!eventsLoading && (!events || events.length < 1) && <NoEvents />}

        {!eventsLoading &&
          !!events &&
          events.length > 0 && (
            <React.Fragment>
              <Grid container={true} spacing={24} direction="row">
                <Hidden xsDown={true}>
                  <Grid item={true} sm={12}>
                    <span className="eventListTitle">Upcoming Events</span>
                    <Link to={'/all-events'} className="eventListShowAll">
                      View all events
                    </Link>
                  </Grid>
                </Hidden>

                <Grid item={true} xs={12} className="eventsRow">
                  <Hidden smUp={true}>
                    <Typography
                      className="eventsListCaption"
                      variant="display1"
                      gutterBottom={true}
                    >
                      Upcoming Events
                    </Typography>
                  </Hidden>
                  <div className="eventsList">
                    {this.renderCarousel(upcomingEvents, false)}
                  </div>
                </Grid>

                <Hidden xsDown={true}>
                  <Grid item={true} sm={12}>
                    <span className="eventListTitle">Upcoming Playlists</span>
                    <Link to={'/all-playlists'} className="eventListShowAll">
                      View all playlists
                    </Link>
                  </Grid>
                </Hidden>

                <Grid item={true} xs={12} className="eventsRow">
                  <Hidden smUp={true}>
                    <Divider />
                    <Typography
                      className="eventsListCaption"
                      variant="display1"
                      gutterBottom={true}
                    >
                      Upcoming Playlists
                    </Typography>
                  </Hidden>
                  {this.renderCarousel(
                    upcomingEvents,
                    true,
                    'No Upcoming Playlists'
                  )}
                </Grid>
              </Grid>
            </React.Fragment>
          )}
      </div>
    )
  }
}

export default Events
