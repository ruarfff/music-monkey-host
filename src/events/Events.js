import React, { Component } from 'react'
import PropTypes from 'prop-types'
import Button from 'material-ui/Button'
import AddIcon from '@material-ui/icons/Add'
import './Events.css'

class Events extends Component {
  render() {
    return (
      <div>
        <Button variant="fab" color="primary" aria-label="add">
          <AddIcon />
        </Button>
      </div>
    )
  }
}

Events.propTypes = {}

export default Events
