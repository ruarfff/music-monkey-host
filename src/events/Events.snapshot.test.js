import React from 'react'
import renderer from 'react-test-renderer'
import Events from './Events'

describe('Events Snapshot', () => {
  it('has not changed unexpectedly', () => {
    const component = renderer.create(<Events />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
