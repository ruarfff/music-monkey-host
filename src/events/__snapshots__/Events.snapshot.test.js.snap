// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Events Snapshot has not changed unexpectedly 1`] = `
<div>
  <button
    aria-label="add"
    className="MuiButtonBase-root-16 MuiButton-root-1 MuiButton-raised-6 MuiButton-fab-11 MuiButton-raisedPrimary-7"
    disabled={false}
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiButton-label-2"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-18"
        color={undefined}
        focusable="false"
        viewBox="0 0 24 24"
      >
        <g>
          <path
            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
          />
        </g>
      </svg>
    </span>
    <span
      className="MuiTouchRipple-root-24"
    />
  </button>
</div>
`;
